sdk/synapse/Azure.Analytics.Synapse.Spark/api/Azure.Analytics.Synapse.Spark.netstandard2.0.cs

```diff
@@ -21,7 +21,9 @@ public partial class SparkBatchOperation : Azure.Operation<Azure.Analytics.Synap
         public override string Id { get { throw null; } }
         public override Azure.Analytics.Synapse.Spark.Models.SparkBatchJob Value { get { throw null; } }
         public override Azure.Response GetRawResponse() { throw null; }
+        public Azure.Response UpdateStatus(bool shouldWaitForJobExectuion, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
```

Comment:
Operation<T>'s `UpdateStatus` method is provided to enable manual polling of the operation -- so customers would typically only use it in advanced scenarios.  The way the LRO pattern is intended to be used for the majority of scenarios is that a customer would call:

```
SparkBatchOperation createOperation = await client.StartCreateSparkBatchJobAsync(options);
SparkBatchJob jobCreateResponse = await createOperation.WaitForCompletionAsync();
```

My recommendation is the following:

1. Create an enum that describes the different ways the operation could complete (instead of the `waitForJobExecution` bool)
2. Add a property to SparkBatchJobOptions letting the caller set the enum value
3. Add a parameter to the SparkBatchOperation constructor, so if a customer wants to resume the operation by creating a new instance of SBOperation, they can set this property there.

Reply:
These are great suggestions! I will update code as suggested.